docs/proposals/odo-debug.md

```diff
@@ -0,0 +1,100 @@
+# odo debug support proposal
+odo debug command is meant for debugging code running as odo component.
+First implementation in odo will just support Java and NodeJS components.
+
+
+Table of Contents:
+- [odo debug support proposal](#odo-debug-support-proposal)
+  - [Design overview](#design-overview)
+    - [New odo commands](#new-odo-commands)
+      - [`odo debug start`](#odo-debug-start)
+    - [port-forward.json format](#port-forwardjson-format)
+      - [`odo debug stop`](#odo-debug-stop)
+      - [`odo debug status`](#odo-debug-status)
+    - [`odo debug port-forward LOCAL_PORT:REMOTE_PORT`](#odo-debug-port-forward-localportremoteport)
+- [Languages support](#languages-support)
+  - [NodeJS](#nodejs)
+    - [Problem 1](#problem-1)
+      - [Solution 1](#solution-1)
+      - [Solution 2](#solution-2)
+  - [Java](#java)
+
+## Design overview
+
+The debug mode inside the container is controlled by `DEBUG_MODE` and `DEBUG_PORT` environment variables.
+- if `DEBUG_MODE == true` than the code should be executed with remote debugging capabilities enabled.
+- `DEBUG_PORT` is optional and controls the port of the remote debugger.
+
+Odo sets up port-forwarding to allow a debugger to connect to the running process. 
+Debugger client will always connect to `localhost:$DEBUG_PORT`
```

Comment:
I'm assuming we'll use a const / default port for odo communication or is this based on each language / image?

Reply:
yes, we will use some default port